core/src/test/java/kafka/server/share/SharePartitionManagerTest.java

```diff
@@ -314,16 +315,16 @@ public void testNewContext() {
 
         // Test trying to create a new session with an invalid epoch
         assertThrows(InvalidShareSessionEpochException.class, () -> sharePartitionManager.newContext(groupId, reqData2, EMPTY_PART_LIST,
-            new ShareRequestMetadata(shareSessionKey2.memberId(), 5), true));
+            new ShareRequestMetadata(shareSessionKey2.memberId(), 5), true, CONNECTION_ID));
```

Comment:
I would have thought that by definition, this would not be the same connection ID as the session which was previously added into the cache.